src/controller/BUILD.gn

```diff
@@ -18,6 +18,9 @@ static_library("controller") {
   output_name = "libChipController"
 
   sources = [
+    "${chip_root}/src/app/server/Mdns.cpp",
```

Comment:
Don't do this.

A source file should belong to **one** library at a time which determines its options (defines, include dirs, etc). If you want to share code, create a new library and make it a dependency.